app/src/androidTest/java/com/nervousfish/nervousfish/test/QRExchangeKeyActivitySteps.java

```diff
@@ -0,0 +1,132 @@
+
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+import org.junit.Rule;
+
+import cucumber.api.CucumberOptions;
+import cucumber.api.java.en.Given;
+import cucumber.api.java.en.Then;
+import cucumber.api.java.en.When;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.intent.Intents.intending;
+import static android.support.test.espresso.intent.matcher.IntentMatchers.toPackage;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static junit.framework.Assert.assertTrue;
+
+
+@CucumberOptions(features = "features")
+public class QRExchangeKeyActivitySteps {
+
+    private final IServiceLocator serviceLocator = (IServiceLocator) BaseTest.accessConstructor(ServiceLocator.class, Instrumentation.filesDir);
+
+    @Rule
+    public ActivityTestRule<QRExchangeKeyActivity> mActivityRule =
+            new ActivityTestRule<>(QRExchangeKeyActivity.class, true, false);
+
+    @Given("^I am viewing QRExchange activity$")
+    public void iAmViewingQRExchangeActivity() {
+
+        final Intent intent = new Intent();
+        intent.putExtra(ConstantKeywords.SERVICE_LOCATOR, this.serviceLocator);
+        mActivityRule.launchActivity(intent);
+
+    }
+
+    @Given("^I am viewing the QRExchange activity$")
+    public void iAmViewingTheQRExchangeActivity() {
+
+        final Intent intent = new Intent();
+        intent.putExtra(ConstantKeywords.SERVICE_LOCATOR, this.serviceLocator);
+        mActivityRule.launchActivity(intent);
+
+    }
+
+
+    @When("^I press the generate button$")
+    public void iPressTheGenerateButton(){
+        onView(withId(R.id.generate_QR_button)).perform(click());
+    }
+
+    @When("^I press the scan button$")
+    public void iPressTheScanButton(){
+        onView(withId(R.id.scan_button)).perform(click());
+    }
+
+    @Then("^I should see a popup with my qr code$")
+    public void iShouldSeeAPopupWithMyQRCode() {
+        AlertDialog dialog = (AlertDialog) BaseTest.getField(mActivityRule.getActivity(), "lastDialog");
+        assertTrue(dialog.isShowing());
+
+        QRExchangeKeyActivity QRExchangeActivity = mActivityRule.getActivity();
+        IKey publicKey = (IKey) BaseTest.getField(QRExchangeActivity, "publicKey");
+        Bitmap qrTest = QRGenerator.encode(publicKey.getType() + " "
+                + publicKey.getName() + " " + publicKey.getKey());
+
+        EspressoTestsMatchers etMatchers = new EspressoTestsMatchers();
+        onView(withId(R.id.QR_code_image)).check(matches(etMatchers.withDrawable(qrTest)));
+    }
+
+    @Then("^I should go to the Barcode Scanner app$")
+    public void iShouldGoToTheBarcodeScannerApp() {
+        intending(toPackage("com.google.zxing.client.android.scan"));
```

Comment:
`intending` is not used to verify anything, you should be using `intended`. See [documentation](https://google.github.io/android-testing-support-library/docs/espresso/intents/)